test/Tests/src/Network/Http_Spec.enso

```diff
@@ -1,332 +1,205 @@
-                  "json": {
-                    "key": "val"
-                  }
-                }
-            json_text = '''
-                { "key": "val" }
-            req = Request.new HTTP_Method.Post url_post
-            req_with_body = req.with_json json_text
-            res = HTTP.new.request req_with_body
-            res.code.should_equal HTTP_Status_Code.ok
-            res.body.decode_as_json.should_equal expected_response
+            response = HTTP.post url_post test_file
+            response.at "data" . should_equal test_file.read_text
+
+        Test.specify "Can perform a Text POST with auto-conversion" <|
+            response = HTTP.post url_post "hello world"
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can skip auto-parse" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") try_auto_parse_response=False
+            response.body.bytes.length . should_equal 383
+
+        Test.specify "Can send a custom header" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") headers=[Header.new "Custom" "asdf"]
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+            response.at "headers" . at "Custom" . should_equal "asdf"
+
+        Test.specify "Cannot POST when output context is disabled" <|
+            Context.Output.with_disabled <|
+                HTTP.post url_post (Request_Body.Text "hello world") . should_fail_with Forbidden_Operation
+
+        Test.specify "Can perform a Request_Body.Text PUT" <|
+            url_put = base_url_with_slash + "put"
+            response = HTTP.post url_put (Request_Body.Text "hello world") method=HTTP_Method.Put
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can perform a Request_Body.Text PATCH" <|
+            url_patch = base_url_with_slash + "patch"
+            response = HTTP.post url_patch (Request_Body.Text "hello world") method=HTTP_Method.Patch headers=[Header.content_type "application/diff"]
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "application/diff"
+
+        Test.specify "Unsupported method" <|
+            url_get = base_url_with_slash + "get"
+            HTTP.post url_get (Request_Body.Text "hello world") method=HTTP_Method.Get . should_fail_with Illegal_Argument
```

Comment:
Why is this failing with `Illegal_Argument`? Shouldn't it be some HTTP related error?

If I understand correctly, the issue here is `405 Method Not Allowed`, right?

Also, we should inspect the `to_display_text` of the error to see if the message is something that we expect and not a random different error.

Reply:
You're right, it should be a Request_Error. I wouldn't say it's a proper HTTP 405 error; it's not that the endpoint doesn't support the method, but that we have provided two methods (`fetch` and `post`) which divide up all the allowed methods, and this test is using the wrong method with the specified HTTP_Method.

Reply:
Oh, I misunderstood the purpose of this test.

I thought that you are hitting the `/get` endpoint and it is responding with 405.

In case of `post` not liking `GET` then I'd stay with `Illegal_Argument`. Let's just inspect its message to ensure it is sane.

Reply:
Related - it feels to me like `DELETE` should be rather allowed in `post` than `fetch`.

My intuition for `fetch` is 'get some data, usually in a non-side-effecting way' - i.e. GET, OPTIONS, HEAD.

And `post` is 'perform some web action'. So IMHO `DELETE` fits `post` much more - it could just require an empty body. Although I think the HTTP standard does not forbid to include a body with DELETE (though it does not make much sense to do so).

If we flip `DELETE` to `post` we can then simplify execution contexts - simply `fetch` could run always and `post` would check if Context.Output is enabled.

For sure `DELETE` cannot run in disabled Output context.

Reply:
Done